src/plugins/embeddable/public/lib/editable_embeddable/navigate_to_editor.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart } from '@kbn/core/public';
+import { apiHasParentApi, apiPublishesSearchSession } from '@kbn/presentation-publishing';
+import { firstValueFrom } from 'rxjs';
+import { EmbeddableStart } from '../..';
+import { apiHasAppContext, NavigateToEditorApi } from './types';
+
+/**
+ * Call this function to navigate to a given editor application
+ * @param services The services needed to perform the navigation
+ * @param api The embeddable API, which should, at minimum, provide a callback for fetching
+ *            the editor app target (app ID, path to navigate to, and generic editor URL)
+ */
+export const navigateToEditor = async (
+  services: { core: CoreStart; embeddable: EmbeddableStart },
+  api: NavigateToEditorApi
+) => {
+  const appTarget = await api.getEditorAppTarget();
+  if (appTarget.editApp && appTarget.editPath) {
+    /** The editor has a dedicated app */
+    const apiContext = apiHasAppContext(api) ? api.getAppContext() : undefined;
+    const parentApiContext =
+      apiHasParentApi(api) && apiHasAppContext(api.parentApi)
+        ? api.parentApi.getAppContext()
+        : undefined;
+    let currentAppId = parentApiContext?.currentAppId ?? apiContext?.currentAppId;
+    if (!currentAppId) {
+      /** If the API is not providing a current app ID, then get it from Core */
+      currentAppId = await firstValueFrom(services.core.application.currentAppId$);
```

Comment:
Do we ever anticipate the `appId` returned from an embeddable to be different than the `currentAppId$` from core? If not, why do we have this extra interface?

Reply:
Good question! I was trying to keep the logic as close as possible to the old `edit_legacy_embeddable` code, and that is what we were doing there - I thought of removing the embeddable context `currentAppId` logic and simply using Core's, but got nervous since this is such old code at this point 🙈 Do you think it's okay to just rely on Core here?

Reply:
I think it's completely okay to rely on Core. I'd do a quick audit to make sure no embeddables behave differently, but if all existing usages work the same it's much simpler to just rely on Core.

Reply:
Talked this through offline and we decided to leave this as-is for now - looks like the `currentAppId` from the `EmbeddableAppContext` type is used in other places (like in Lens, for example), so we would have to modify quite a few files to remove `currentAppId` from this type.